pkg/collector/corechecks/system/disk/diskv2/disk.go

```diff
@@ -0,0 +1,623 @@
+	UseLsblk             bool              `yaml:"use_lsblk"`
+	BlkidCacheFile       string            `yaml:"blkid_cache_file"`
+	ServiceCheckRw       bool              `yaml:"service_check_rw"`
+	CreateMounts         []mount           `yaml:"create_mounts"`
+	DeviceInclude        []string          `yaml:"device_include"`
+	DeviceWhitelist      []string          `yaml:"device_whitelist"`
+	DeviceExclude        []string          `yaml:"device_exclude"`
+	DeviceBlacklist      []string          `yaml:"device_blacklist"`
+	ExcludedDisks        []string          `yaml:"excluded_disks"`
+	ExcludedDiskRe       string            `yaml:"excluded_disk_re"`
+	FileSystemInclude    []string          `yaml:"file_system_include"`
+	FileSystemWhitelist  []string          `yaml:"file_system_whitelist"`
+	FileSystemExclude    []string          `yaml:"file_system_exclude"`
+	FileSystemBlacklist  []string          `yaml:"file_system_blacklist"`
+	ExcludedFileSystems  []string          `yaml:"excluded_filesystems"`
+	MountPointInclude    []string          `yaml:"mount_point_include"`
+	MountPointWhitelist  []string          `yaml:"mount_point_whitelist"`
+	MountPointExclude    []string          `yaml:"mount_point_exclude"`
+	MountPointBlacklist  []string          `yaml:"mount_point_blacklist"`
+	ExcludedMountPointRe string            `yaml:"excluded_mountpoint_re"`
+	DeviceTagRe          map[string]string `yaml:"device_tag_re"`
+}
+
+func sliceMatchesExpression(slice []regexp.Regexp, expression string) bool {
+	for _, regexp := range slice {
+		if regexp.MatchString(expression) {
+			return true
+		}
+	}
+	return false
+}
+
+// Check represents the Disk check that will be periodically executed via the Run() function
+type Check struct {
+	core.CheckBase
+	initConfig          diskInitConfig
+	instanceConfig      diskInstanceConfig
+	includedDevices     []regexp.Regexp
+	excludedDevices     []regexp.Regexp
+	includedFilesystems []regexp.Regexp
+	excludedFilesystems []regexp.Regexp
+	includedMountpoints []regexp.Regexp
+	excludedMountpoints []regexp.Regexp
+	deviceTagRe         map[*regexp.Regexp][]string
+	deviceLabels        map[string]string
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
+	}
+	if c.instanceConfig.TagByLabel {
+		err = c.fetchAllDeviceLabels()
+		if err != nil {
+			log.Debugf("Unable to fetch device labels: %s", err)
+		}
+	}
+	err = c.collectPartitionMetrics(sender)
+	if err != nil {
+		return err
+	}
+	err = c.collectDiskMetrics(sender)
+	if err != nil {
+		return err
+	}
+	sender.Commit()
+
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := c.CommonConfigure(senderManager, initConfig, data, source)
+	if err != nil {
+		return err
+	}
```

Comment:
In the Python check we check that we do not configure more than one instance: 
```
if instances is not None and len(instances) > 1:
            raise ConfigurationError('Disk check only supports one configured instance.')
```
https://github.com/DataDog/integrations-core/blob/22f0280a30d8c3f3626feb0563a95c0f3b87203a/disk/datadog_checks/disk/disk.py#L41-L42

Is this something we should do in the Go version as well?